uint8 dev_class
uint8 dev_type
uint8 dev_number
uint8 dev_enable
